src/app/main.py

```
@@ -0,0 +1,102 @@
+
+APPLICATION = FastAPI(title="University Marks Manager API", lifespan=lifespan)
+
+
+APPLICATION.include_router(api, prefix="/api")
+APPLICATION.include_router(views)
+
+# Enable server-side sessions for lightweight state (e.g., selected course)
+# NOTE: Replace the secret key with an environment variable for production use.
+APPLICATION.add_middleware(
+    SessionMiddleware,
+    secret_key="dev-secret-change-me",
```

Ruff (0.14.3)

Possible hardcoded password assigned to argument: "secret_key"

(S106)